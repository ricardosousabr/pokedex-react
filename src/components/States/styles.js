import styled from "styled-components";

export const Container = styled.div`
  color: black;
  display: flex;
  justify-content: space-around;
  flex-direction: column;
`;
