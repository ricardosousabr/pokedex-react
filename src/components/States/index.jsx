import PokeState from "../PokeState";
import { Container } from "./styles";

export default function States({ statesPokemon }) {
  const statesPoke = statesPokemon.stats;
  const primarySkill = statesPokemon.abilities[0].ability.name;
  const secondarySkill = statesPokemon.abilities[1].ability.name;

  return (
    <Container>
      <div>
        <p>Primary skill: {primarySkill}</p>
        {secondarySkill ? (
          <p>Secondary skill: {secondarySkill}</p>
        ) : (
          <p>no secondary skill</p>
        )}
      </div>
      <div>
        {statesPoke.map(({ base_stat, stat }) => (
          <PokeState
            key="satespoke"
            stateName={stat.name}
            stateValue={base_stat}
          />
        ))}
      </div>
    </Container>
  );
}
