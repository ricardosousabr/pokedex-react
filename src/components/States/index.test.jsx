/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from "@testing-library/react";
import States from "./index";

const mock = {
  stats: [
    {
      base_stat: 35,
      stat: {
        name: "hp",
      },
    },
  ],
  abilities: [
    {
      ability: {
        name: "static",
      },
    },
    {
      ability: {
        name: "tttt",
      },
    },
  ],
};
it("renders homepage unchanged", () => {
  jest.mock("./index");
  const { container } = render(<States statesPokemon={mock} />);
  expect(container).toMatchSnapshot();
});
