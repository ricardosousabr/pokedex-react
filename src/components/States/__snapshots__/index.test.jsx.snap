// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="styles__Container-sc-1p1aihv-0 SCoIR"
  >
    <div>
      <p>
        Primary skill: 
        static
      </p>
      <p>
        Secondary skill: 
        tttt
      </p>
    </div>
    <div>
      <div>
        <p>
          hp
          :
          35
        </p>
        <progress
          class="styles__BarStatus-sc-1e9ujot-0 eKRLMZ"
          max="100"
          value="35"
        />
      </div>
    </div>
  </div>
</div>
`;
